pyccel/ast/builtin_methods/set_methods.py

```diff
@@ -132,3 +132,30 @@ def __init__(self, set_variable):
         self._precision = set_variable._precision
         self._class_type = set_variable._class_type
         super().__init__(set_variable)
+
+
+class SetPop(SetMethod):
+    """
+    Represents a call to the .pop() method.
+
+    The pop() method pops an element from the set. 
+    It does not take any argument but returns the popped 
+    element. It raises an error if the element is not 
+    present in the set.
```

Comment:
This docstring contains a lot more about the `pop()` method than the `SetPop`class that it is documenting. It could be good to relate the two. E.g. by also mentioning how/if the `SetPop`class handles the mentioned error

Reply:
Yeah, you're right, but this class doesn't contain any error checks. I always have a question about why the docstring doesn't match what the class does?

Reply:
> I always have a question about why the docstring doesn't match what the class does?

It is a little worrying if you think this is the case. The docstring _should_ describe what the class does. If you don't think this is the case then you may need to improve your docstrings.

In the AST, the class exists to describe a method so the two are linked. It does therefore make sense to describe the method being represented, but docs for these methods already exist. So you should only need to describe the method (e.g. `pop()`) briefly to give context to understand how the class works.

> Yeah, you're right, but this class doesn't contain any error checks.

That is why I think it is worth mentioning as you might expect it to given the docstring. For example you could write:
```
    Represents a call to the .pop() method.

    The pop() method pops an element from the set. 
    It does not take any arguments but returns the popped 
    element. It raises an error if the set is empty.
    The class does not raise an error as it assumes that the
    user code is valid.
```

Reply:
Understood, I'll make the changes to the docstring according to your suggestion.

Reply:
Fixed : d910add